src/convertToObject.js

```diff
@@ -5,8 +5,23 @@
  *
  * @return {object}
  */
-function convertToObject(sourceString) {
-  // write your code here
+function convertToObject(stringForConvert) {
+  const result = stringForConvert.trim().split(';');
+
+  const resultObject = result
+    .filter((element) => {
+      if (element.includes(':')) {
+        return element;
+      }
+    })
+    .map((element) => {
+      return element.split(':');
+    })
+    .reduce((object, element) => {
+      return { ...object, [element[0].trim()]: element[1].trim() };
```

Comment:
The 'reduce' method's callback function can be simplified. There's no need to return a new object on each iteration. Instead, you can directly assign the new property to the 'object' and return it. This avoids unnecessary object spread operations, which can be costly in terms of performance.